cmake/SetupPackages.cmake

```diff
@@ -27,4 +27,35 @@ if (ENABLE_TBB)
     message(WARNING "TBB NOT FOUND")
     set(ENABLE_TBB Off)
   endif()
-endif ()
\ No newline at end of file
+endif ()
+
+if (ENABLE_CUDA OR ENABLE_EXTERNAL_CUB)
+  find_package(CUB)
+  if (CUB_FOUND)
+    blt_register_library(
+      NAME cub
+      INCLUDES ${CUB_INCLUDE_DIRS})
+  else()
+    if (ENABLE_EXTERNAL_CUB)
```

Comment:
Is there a good reason why the logic is different for cub and rocprim here and below?

Reply:
We still have a rocprim submodule so we don't use find_package for that. I split up the error messages to be more useful and informative depending on the setting.

Reply:
OK. Why do we want to keep the rocprim submodule?

Reply:
I'll check if rocprim is included in the rocm install, but I don't remember seeing it in there.

Reply:
Not a big deal. We should document the difference between how we handle cub and rocprim in our user guide.

Reply:
It appears that rocprim is part of the rocm install after all. I'll change the code to use that instead of the submodule.

Reply:
done